### PR TITLE
ci(deps): bundle every Lambda handler when dependencies change

### PR DESCRIPTION
**Pull request by poysama**
## The gap

A dependency bump broke every staging deploy for about six hours: a root `pnpm.overrides` pin held `@smithy/core` below the floor its consumers declared, and esbuild failed to bundle every Lambda inside `sst deploy`. #2914 fixed the pin. The overrides guard on #2927 closes that one shape (an override contradicting a declared range). It does not close the class: a package dropping an export, an ESM/CJS interop change, a native binary that will not resolve.

Nothing in CI runs esbuild over the Lambda handlers, so the first thing to bundle them after a dependency change is a real deploy. `sst-infra-preview-gate.yml` forces a preview for `infra/`, `b4m-core/infra/`, `sst.config.ts` and changed `sst` pins, but a `pnpm-lock.yaml` change does not trip it. So a dependency PR merges, main pushes, and staging is where the bundle is first attempted.

## Why this is not just `sst build`

`sst build` is not a command. sst 4.17.1's command list is `init / dev / deploy / diff / add / install / secret / shell / remove / unlock / version / upgrade / telemetry / refresh / state / cert / tunnel / diagnostic`.

`sst diff` is the closest thing, and it cannot run credential-free. With no credentials it stops before evaluating anything:

```
AWS credentials are not configured. Try configuring your profile in `~/.aws/config` ...
aws: failed to refresh cached credentials, no EC2 IMDS role found
```

With credentials that do not authenticate, it gets one step further and fails on the state passphrase, still before a single resource is evaluated and long before any handler is bundled:

```
aws: operation error SSM: GetParameter, ... api error UnrecognizedClientException
```

`sst install` does work credential-free (it builds `sst.config.ts` with esbuild and downloads providers), but it never calls `run()`, so no Function is ever constructed and no handler is ever bundled. The bundling itself happens in `Runtime.Build`, an RPC into sst's Go CLI that only the deploy path reaches.

So there is no credential-free sst path to a handler bundle. The second option in the brief it is.

## What this does

`scripts/check-lambda-bundle.mjs` reads the handler entry points `infra/` declares and runs one esbuild pass over them with sst's externals.

- **Entry points come from `infra/`**, not a hand-maintained list, so a new Lambda joins the guard by existing. All 105 declared handlers are plain string literals today, and the script fails loudly if one ever stops being one rather than silently dropping a Lambda.
- **102 of the 105 bundle here.** The other three resolve under `apps/client/server/premium-generated/`, where the source arrives with an overlay. Those are skipped by path, and the skip is narrow: a missing handler file anywhere else is a hard failure, and an overlay handler whose source *is* hydrated gets bundled like any other.
- **Externals are read out of `sst.config.ts`**, not copied. A fourth entry added to `ALWAYS_EXTERNAL` reaches the guard on its own, and if that declaration is ever renamed the guard stops with a message saying so instead of reddening every handler that imports a native addon.
- **Code splitting is what makes it affordable.** Without it esbuild emits a standalone bundle per handler and spends about three minutes printing 2.5 GB of duplicated output. With it, the same 9,902 modules resolve and link in about 1.5 seconds. `distinctInputs` is identical either way, which is the part the guard actually reads.
- **Nothing is written.** `write: false`, and the `outdir` esbuild needs for path computation points under `node_modules/` so even a debug run with `write: true` cannot dirty the tree.

## Verification

Restoring the pre-fix pin (`@smithy/core` 3.34.1 to 3.29.0, `@smithy/types` 4.18.0 to 4.15.1) and running the shipped script reproduces the outage in 1.0s, exit 1:

```
ERROR: 8 distinct bundling failure(s) across 102 handlers (1.0s):

  No matching export in ".../@smithy/core/dist-es/submodules/serde/index.js" for import "hasOwn"
    at .../@smithy/node-http-handler/dist-es/get-transformed-headers.js:1
  ...
```

On the pins as they stand:

```
Bundling 102 Lambda handlers declared in infra/
Skipping 3 overlay-only handlers (source arrives with the premium overlay)
External: sharp, pg-native, isolated-vm, @huggingface/transformers, onnxruntime-node

OK: 102 handlers bundled, 9902 modules resolved, 1.4s
```

16 unit tests cover the extraction, the overlay split, the externals parse and the message dedupe. Two of them assert against the live `infra/` and `sst.config.ts`, so drift in either one reddens rather than quietly narrowing the guard.

## Cost

Measured on this PR's own first run (cold pnpm cache, no turbo cache), 2m00s wall clock end to end:

| step | wall clock |
|---|---|
| checkout + pnpm + node setup | 11s |
| `pnpm install --frozen-lockfile` | 29s |
| `pnpm turbo:core:build` | 60s |
| `node scripts/check-lambda-bundle.mjs` | 4s |
| unit tests | 1s |
| cache save + teardown | 9s |

The core build is the bulk of it and it is unavoidable: every handler reaches `@bike4mind/*` through a `dist/` entry point, so an unbuilt core package is an unresolvable import rather than a working guard. A real deploy builds core first for the same reason. Later runs should come in under this as the pnpm store cache warms.

The alternative, widening the preview gate to `pnpm-lock.yaml`, is a ~20 minute throwaway preview deploy on every dependency PR, which Dependabot volume makes punishing.

## Fidelity, and what this does not cover

This replicates sst's bundling step rather than calling it. sst builds each Function separately through its Go CLI with esbuild v0.27.2 embedded; this runs one pass with the repo's own pin (0.28.1, which the root overrides already fix for the whole tree). What carries over is module resolution and ESM/CJS linking, which is where a dependency change breaks the bundle. What does not carry over is sst-specific packaging: `nodejs.install` layout, the per-Function output shape, the Next.js/OpenNext site build.

Two other known limits, both deliberate:

1. The guard applies only the global externals from `sst.config.ts`, not the per-Function `nodejs.install` / `nodejs.esbuild.external` in `infra/mcp.ts:39` and `infra/queues.ts:1083`. That means it bundles *more* than a real deploy does, which is more resolution coverage. It cuts the other way too, and the body originally undersold this: a package that only works kept external would fail here and deploy fine. That is the false red to expect. Everything bundles clean today, and the comment on `parseAlwaysExternal` names both call sites and says which list to add to when it happens.
2. It is path-filtered, so it is not eligible to be a required check. A filtered workflow reports nothing on a PR it skips, and a required check that never reports hangs the PR pending. Making it required means moving the applicability decision inside the job the way `sst-infra-preview-gate.yml` does. It runs on `merge_group` already, so it reports there and can be promoted without a chicken-and-egg problem.

An obvious follow-up, left out to keep this in scope: widening the trigger to `infra/**` and `apps/client/server/**` would also catch a typo'd or deleted handler path in 1.5 seconds rather than on a preview deploy.

## Review round 1

Fixed:

- **`collectHandlers` could not tell a comment from a declaration.** Verified both directions against the function: a comment containing `handler:` threw "declares a handler that is not a string literal" with a made-up diagnosis, and a comment shaped like a literal was silently extracted as a real entry point. It scans every file under `infra/` on every dependency PR, so that was a repo-wide CI break waiting on a future comment.

  The suggested fix was a `^\s*//` line filter. That is not enough here: two of the three existing `Handler:` mentions in `infra/` are JSDoc (`infra/emailIngestion.ts:148` and `:305` are ` * Handler:`), which a line filter does not strip, and it does not handle a trailing comment either. So comments now go through `esbuild.transformSync` instead, which is already the tool doing the bundling and handles JSDoc, trailing comments and `//` inside a string literal alike. Proof against the real files: lowercasing every `Handler:` mention in `infra/queues.ts` and `infra/emailIngestion.ts` leaves the handler count unchanged, where before it hard-failed.

- **Word boundary on both patterns.** `retryHandler:` was never actually matched (the pattern is case-sensitive and that is a capital H), but `myhandler:` was. `(?<![A-Za-z])` on both closes it, with a test.

- **Path filter now covers the guard itself** (`scripts/check-lambda-bundle.mjs`, its test file, and the workflow), with the reason inline. Confirmed nothing else reaches those tests: root `scripts/` is not a workspace package, `ci.yml` runs only `--dir infra` and `--dir scripts/codemods`, and `audit-gate.yml` filters to `audit-gate`.

- **End-to-end failure-path coverage.** The esbuild pass is now an exported `bundleEntryPoints`, with tests for a resolving graph, an unresolvable import, an external suppressing that import, and the dedupe shape. 26 tests, up from 16.

- **`SST_BUILTIN_EXTERNAL` provenance.** Correct that it is not readable from the vendored JS package, since the bundling lives in sst's Go binary. It came off that binary's string table, where `sharp` and `pg-native` sit alongside the other esbuild option names it sets (`keepNames`, `sourcemap`, `OnResolve`). Noted in the comment so the next reader does not have to redo it.

## Note on the root dependency

`esbuild` moves into root `devDependencies` at `0.28.1`, the version root `pnpm.overrides` already pins for the whole tree, so the lockfile delta is one importer entry. It was previously resolvable from root only as a hoisting accident via `apps/client`. Same precedent as the `semver` addition on #2927.

### CODE REVIEW

File: .github/workflows/lambda-bundle-guard.yml [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+# Path-filtered, so it is NOT eligible to be a required check: a filtered
+# workflow reports nothing on a PR it skips, and a required check that never
+# reports hangs the PR pending. Making it required means moving the applicability
+# decision inside the job, the way sst-infra-preview-gate.yml does.
+
+on:
+  pull_request:
+    branches-ignore: [prod]
+    types: [opened, synchronize, reopened, ready_for_review]
+    paths:
```
Comment (vinchi777) [inline .github/workflows/lambda-bundle-guard.yml:28]: P2: the filter covers what *triggers* a bundle break but not the guard itself. `scripts/check-lambda-bundle.mjs` and `scripts/__tests__/check-lambda-bundle.test.mjs` are not in this list, and nothing else in CI picks them up: root `scripts/` is not a workspace package (so `pnpm -r test` skips it), `ci.yml` runs only `--dir infra` and `--dir scripts/codemods`, and `audit-gate.yml` filters its run to `audit-gate`. So a PR that edits the guard or its tests gets zero signal, and the 16 tests that are the reason to trust this thing only run on PRs that happen to touch a manifest.

Adding `scripts/check-lambda-bundle.mjs` and `scripts/__tests__/check-lambda-bundle.test.mjs` to `paths` closes it, at the cost of one extra ~2m run on the rare PR that touches them.

File: scripts/check-lambda-bundle.mjs [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+const OVERLAY_HANDLER_DIR = 'premium-generated/';
+
+const SOURCE_EXTENSIONS = ['.ts', '.tsx', '.mts', '.cts', '.js', '.mjs', '.cjs', '.jsx'];
+
+// Every handler in infra/ is a plain string literal of the form
+// `path/to/file.exportName`. collectHandlers asserts that stays true.
+const HANDLER_PATTERN = /handler:\s*(['"`])([^'"`\n]+)\1/g;
+
+// sst.config.ts registers these as external on EVERY Function via $transform.
+// Read rather than copied: a fourth entry added there has to reach this guard,
```
Comment (vinchi777) [inline scripts/check-lambda-bundle.mjs:51]: P3: the PR body says "the script comment says where" to add a per-Function external when one later becomes unbundlable, but this comment only covers `ALWAYS_EXTERNAL`. There is nothing here pointing at `infra/mcp.ts:39` (`external: ['@bike4mind/mcp', '@bike4mind/common']`) or `infra/queues.ts:1083` (`external: ['isolated-vm']`, already covered by `ALWAYS_EXTERNAL`).

The divergence is in the safe direction today, but the person who eventually hits a false red from it will be reading exactly this comment. One line naming those two call sites would make the claim true.

File: scripts/check-lambda-bundle.mjs [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+// Returns [{ handler, declaredIn }], deduped, in declaration order.
+export function collectHandlers(sources, readFile = f => fs.readFileSync(f, 'utf8')) {
+  const seen = new Map();
+  for (const file of sources) {
+    const source = readFile(file);
+
+    // A handler built from a variable would be invisible to the pattern above and
+    // would drop a Lambda out of this guard with nothing to show for it.
+    const literalCount = [...source.matchAll(HANDLER_PATTERN)].length;
+    const totalCount = [...source.matchAll(/handler:/g)].length;
```
Comment (vinchi777) [inline scripts/check-lambda-bundle.mjs]: P3: `/handler:/g` has no word boundary and no comment awareness, so it counts any occurrence of the substring. A future `// handler: see the queue below` in an `infra/` file makes `totalCount` exceed `literalCount` and hard-fails the guard on a PR that has nothing to do with it; conversely a `retryHandler: 'x'` key would be picked up by `HANDLER_PATTERN` and bundled as if it were an entry point.

Nothing in `infra/` trips it today (the three `Handler:` mentions at `infra/emailIngestion.ts:148`, `:305` and `infra/queues.ts:1313` are capitalised and so miss both regexes), and the error message is clear enough to diagnose. `(?<![A-Za-z])handler:` on both patterns would remove the second direction cheaply.

**Review by vinchi777 — DISMISSED** [review]
| Severity | Finding | Location |
|----------|---------|----------|
| P2 | Path filter omits the guard's own script and tests, so a change to either lands with no CI signal | `.github/workflows/lambda-bundle-guard.yml:28` |
| P3 | Per-Function `nodejs.esbuild.external` is not reflected and the script carries no pointer to it, contrary to the PR body | `scripts/check-lambda-bundle.mjs:46` |
| P3 | `/handler:/g` counts substrings and comments, so unrelated text hard-fails the guard | `scripts/check-lambda-bundle.mjs:83` |

The approach checks out. Replicating sst's bundling step is the right call given `sst install` never calls `run()` and so never constructs a Function, and the esbuild options match what sst's node runtime uses (`format: esm`, `mainFields: ['module','main']`, `keepNames`, `platform: node`) -- nothing under `infra/` overrides `nodejs.format`, so every deployed handler really is ESM. `splitting: true` only collapses duplicate diagnostics; it cannot hide a distinct resolution or linking error, and `metafile.inputs` is unchanged either way. The external list bundles strictly *more* than a deploy does (`infra/mcp.ts` externalizes `@bike4mind/mcp` / `@bike4mind/common` purely for runtime resolution), so the divergence is extra coverage, not a blind spot. Failure modes are guarded where it matters: a non-literal handler throws rather than silently dropping a Lambda, a missing source outside `premium-generated/` is a hard error, an empty entry list fails instead of reporting a vacuous pass, and the live-`infra/` assertions (`missing === []`, `entries.length > 50`) make drift redden rather than quietly narrow the guard. The workflow mirrors `audit-gate.yml` step for step, so it sits inside an existing pattern.

CI is red on `Run Tests (misc)`, but the single failure is `packages/scripts` `20260914000000_ensure-fabfilechunk-residency-index.integration.test.ts` asserting a pre-existing index is absent -- unrelated to this diff.

File: scripts/check-lambda-bundle.mjs [hunk 4]
```diff
@@ -0,0 +1,235 @@
...
+    if (literalCount !== totalCount) {
+      throw new Error(
+        `${file} declares a handler that is not a string literal. This guard finds entry points by reading those literals, so a computed handler is a Lambda it cannot bundle. Either keep the literal or add the entry point to this guard explicitly.`
+      );
+    }
+
+    for (const [, , handler] of source.matchAll(HANDLER_PATTERN)) {
+      if (!seen.has(handler)) seen.set(handler, file);
```
Comment (onoya) [inline scripts/check-lambda-bundle.mjs]: This can't tell a comment from a declaration. `HANDLER_PATTERN` (line 43) and the plain `/handler:/g` count both scan raw file text, so a comment like `// TODO: fix the handler: naming here` bumps `totalCount` without matching `HANDLER_PATTERN` — and this throws "declares a handler that is not a string literal," a hard CI failure with a made-up diagnosis. Worse, a comment that happens to *look* like a literal (`// example: handler: 'docs/foo.bar'`) matches `HANDLER_PATTERN` too, so it gets added to the real handler list and then fails downstream as a "missing" handler once nothing resolves that fake path.

I ran both cases directly against this function: a comment-only mention throws the "not a string literal" error, and a comment shaped like a literal gets silently extracted as a real handler entry.

This matters because it scans every file under `infra/` on every dependency PR, not just the files a given PR touches — "handler" is an extremely common word in exactly this codebase (105 real declarations already), so a future comment anywhere in `infra/` breaks CI for someone else's unrelated PR, and the error message points them at the wrong cause entirely.

Fix: strip comment lines before either the count or the extraction loop runs, e.g.
```js
const source = readFile(file)
  .split('\n')
  .filter(line => !/^\s*\/\//.test(line))
  .join('\n');
```
then use that for both `literalCount`/`totalCount` and the `for (const [, , handler] of source.matchAll(HANDLER_PATTERN))` loop below. Doesn't catch a trailing same-line comment after a real declaration, but that's a much narrower case worth its own test if you want to go further.

**Review by onoya — CHANGES_REQUESTED** [review]
This is a genuinely well-reasoned CI guard — the investigation into why `sst diff` can't run credential-free, and the repro against the actual `@smithy/core` outage pin, are the kind of legwork that makes a review easy to trust. One real bug before this can merge, though: the handler-literal classifier in `collectHandlers` can't tell a code comment from a declaration, so a comment anywhere in `infra/` that happens to contain the word "handler:" will false-fail this check on a totally unrelated PR (details inline). Once that's fixed I'm happy to see this land — everything else below is non-blocking.

Non-blocking, for when you get to it:
- `.github/workflows/lambda-bundle-guard.yml:28-31` — the trigger paths don't include `scripts/check-lambda-bundle.mjs`, `scripts/__tests__/**`, or the workflow file itself. A PR that only edits the guard (say, fixing the bug below) won't run its own tests in CI. Worth adding those three paths.
- `infra/mcp.ts:39` — `@bike4mind/mcp`/`@bike4mind/common` are kept external per-Function in prod, but the guard's global external list doesn't cover them, so it inlines them. Harmless today, but worth a line in the fidelity notes acknowledging this could also produce a false *failure* if either package ever contains something that only works kept external — not just "more coverage."
- No test exercises `main()`'s actual esbuild-failure path end to end (dedupe + exit code) — all 16 tests cover the pure helpers. A tiny fixture with a genuinely unresolvable import would lock in the integration too.
- `SST_BUILTIN_EXTERNAL = ['sharp', 'pg-native']` (line 22) — couldn't verify this against sst's source (the bundling logic lives in its Go binary, not the vendored JS package), so flagging as unverified rather than confirmed.

Happy to file issues for any of these if you want them tracked — just say so.

File: scripts/check-lambda-bundle.mjs (hunk 4 above)
Comment (poysama) [reply scripts/check-lambda-bundle.mjs]: Confirmed both directions against the function before changing anything: a comment mentioning `handler:` threw the "not a string literal" error, and `// example: handler: 'docs/foo.bar'` came back as a real entry in the handler list. Fixed in 5ed5df7b.

I went with a tokenizer rather than the line filter, because the line filter is not enough for this repo specifically. Two of the three existing `Handler:` mentions under `infra/` are JSDoc:

```
infra/emailIngestion.ts:148: * Handler: apps/client/server/emailIngestion/emailParser.dispatch
infra/emailIngestion.ts:305: * Handler: apps/client/server/emailIngestion/emailAnalyzer.dispatch
infra/queues.ts:1313:// Handler:        apps/client/server/queueHandlers/agentExecutor.ts
```

`^\s*//` strips the third and leaves the first two, so lowercasing either of those still breaks CI. It also misses the trailing-comment case you flagged as narrower.

So comments now go through `esbuild.transformSync({ loader: 'ts', legalComments: 'none' })`, which is already the tool doing the bundling. It handles JSDoc, trailing comments, and leaves `//` inside a string literal alone, which a regex strip would have mangled.

Proof against the real files rather than a fixture . lowercasing every `Handler:` mention in both files and re-running:

```
handlers before: 30 | after lowercasing every comment mention: 30
```

Before the fix that combination hard-failed. Six new tests cover line comment, JSDoc, trailing comment, comment-shaped-like-a-literal, `//` inside a string, and the suffixed-key case.

File: scripts/check-lambda-bundle.mjs (hunk 3 above)
Comment (poysama) [reply scripts/check-lambda-bundle.mjs]: Checked the second direction before fixing: `retryHandler:` is not actually picked up, because both patterns are case-sensitive and that is a capital H. `myhandler:` is, though, so the boundary is worth having . added `(?<![A-Za-z])` to both patterns in 5ed5df7b with a test for the lowercase-prefix case.

The comment-awareness half turned out to be a live bug rather than a future one, so it is fixed in the same commit via `esbuild.transformSync` . details on the other thread. Worth noting the `Handler:` mentions you found only miss by capitalisation, so they were one lowercase edit away from breaking CI for an unrelated PR.

File: scripts/check-lambda-bundle.mjs (hunk 2 above)
Comment (poysama) [reply scripts/check-lambda-bundle.mjs:51]: Fair . the body made a claim the code did not back. The comment on `parseAlwaysExternal` now names both call sites (`infra/mcp.ts:39` and `infra/queues.ts:1083`, noting the latter repeats what `ALWAYS_EXTERNAL` already covers) and says which list to add to.

It also now states the failure direction honestly. The original framing of "strictly more coverage" was one-sided: inlining a package that only works kept external would fail here and deploy fine, and that is the false red someone will eventually hit. Body updated to match.

File: .github/workflows/lambda-bundle-guard.yml (hunk 1 above)
Comment (poysama) [reply .github/workflows/lambda-bundle-guard.yml:28]: Verified the gap the same way you described it . nothing else in CI reaches those tests. Root `scripts/` is not a workspace package so `pnpm -r test` skips it, `ci.yml` runs only `--dir infra` and `--dir scripts/codemods`, and `audit-gate.yml` filters its own run to `audit-gate`. So the 16 tests only ran on PRs that happened to touch a manifest, including the one that just fixed a real bug in the guard.

Added `scripts/check-lambda-bundle.mjs`, `scripts/__tests__/check-lambda-bundle.test.mjs` and the workflow file itself to `paths` in 5ed5df7b, with the reason inline so it does not get trimmed later. Test count is 26 now, since the same round added direct coverage of the esbuild failure path.

**Comment by poysama** [comment]
Round 1 addressed in 6ec14ee5b (rebased onto main, which has since picked up the overrides guard . both guards green together locally).

**Blocking:** the comment-vs-declaration bug is real and I reproduced both directions before touching it. Fixed with `esbuild.transformSync` rather than the suggested `^\s*//` line filter, because that filter is insufficient for this repo: two of the three existing `Handler:` mentions under `infra/` are JSDoc (` * Handler:` at `infra/emailIngestion.ts:148` and `:305`), which a line filter leaves in place. Proof against the real files . lowercasing every `Handler:` mention in `infra/queues.ts` and `infra/emailIngestion.ts` now leaves the handler count at 30, where before that combination hard-failed.

**Non-blocking, all four done rather than tracked:**

1. Path filter now covers `scripts/check-lambda-bundle.mjs`, its test file, and the workflow itself, with the reason inline.
2. Fidelity note corrected. "Strictly more coverage" was one-sided . inlining a package that only works kept external would fail here and deploy fine. The comment on `parseAlwaysExternal` now names `infra/mcp.ts:39` and `infra/queues.ts:1083` and says which list to add to.
3. The esbuild pass is exported as `bundleEntryPoints`, with tests for a resolving graph, an unresolvable import, an external suppressing it, and the dedupe shape. 26 tests, up from 16.
4. `SST_BUILTIN_EXTERNAL` provenance: correct that it is not in the vendored JS package, since the bundling lives in the Go binary. It came off that binary's string table, where `sharp` and `pg-native` sit next to the other esbuild option names it sets (`keepNames`, `sourcemap`, `OnResolve`). Recorded in the comment so nobody has to redo it.

Word boundary added too: `retryHandler:` was never actually matched (both patterns are case-sensitive, capital H), but `myhandler:` was, so `(?<![A-Za-z])` is on both now with a test.

Nothing left to file as an issue from this round. On the red `Run Tests (misc)` leg . the failing assertion is in `packages/scripts`, which this branch does not touch at all; the diff is three files under `.github/workflows/` and `scripts/`.

File: scripts/check-lambda-bundle.mjs (hunk 4 above)
Comment (onoya) [reply scripts/check-lambda-bundle.mjs]: Verified — comments now go through `esbuild.transformSync` before either regex runs (`stripComments`, scripts/check-lambda-bundle.mjs:94-97), with `it.each` tests covering a line comment, JSDoc, a trailing comment, and a comment shaped like a literal. Also re-ran it myself against a fresh edge case (an `_handler:` key) and confirmed that one's still open — noted as a follow-up in the review, not blocking. Resolving; reopen if I've read this wrong.

**Review by onoya — APPROVED** [review]
LGTM! 🚀

Reran the classifier's own edge cases myself rather than just reading the diff — the comment-vs-declaration fix (`esbuild.transformSync` in `stripComments`, scripts/check-lambda-bundle.mjs:94) genuinely closes what it says it closes, and the two round-1 blockers (mine and vinchi777's) both check out against the current file. CI's fully green including the new job itself.

One thing I found while poking at it further: `HANDLER_PATTERN`'s word-boundary fix (`scripts/check-lambda-bundle.mjs:47-48`) still misfires on an underscore-prefixed key — I ran `collectHandlers` against `{ _handler: someFn }` next to a real handler and it throws the same "not a string literal" false-positive that round 1 fixed for `myhandler:`. Nothing in `infra/` triggers this today, so it's not blocking, but happy to file it as an issue if you want it tracked (same one-line fix: widen the lookbehind to `(?<![A-Za-z0-9_])`).
